dali/kernels/signal/moving_mean_square.cc

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/kernels/signal/moving_mean_square.h"
+
+namespace dali {
+namespace kernels {
+namespace signal {
+
+namespace {
+template<typename T>
+float Power2(const T &val) {
```

Comment:
`Square` or `sqr`? `Power2` looks waay too much like "power of two". And the result type should be configurable, i.e.:
```
template <typename Result = float, typename T>
Result Square(const T &val) {
  Result x = val;
  return x * x;
}
```

Reply:
Done